python/lsst/ts/IntegrationTests/base_script.py

```diff
@@ -139,6 +139,15 @@ async def wait_for_done(self, data: salobj.BaseMsgType) -> None:
         data : ``lsst.ts.salobj.BaseMsgType``
             The object returned by the ScriptQueue Script Event (evt_script).
         """
+        if (
+            data.scriptState == ScriptState.UNKNOWN
```

Comment:
I would make a list of the "don't care" states like you did with the terminal states. Then this check can just be `data.scriptState in <list of states>`.